package-manager-publish-r/action.yml

```diff
@@ -0,0 +1,87 @@
+name: Publish R Package to Posit Package Manager
+author: Posit Package Manager Team
+inputs:
+  url:
+    description: Package Manager URL
+    required: true
+  api-token:
+    description: API token generated by Package Manager
+    required: true
+  r-version: 
+    description: R version to use
+    default: 'release'
+  source:
+    description: Package Manager Source to publish to
+    required: true
+  distro: 
+    description: Package Manager Distribution to upload binary
+  check:
+    description: Run package checks before publishing
+    default: true
+  replace:
+    description: If version already exists, replace it with new version
+    default: true
+  dir:
+    description: Package source directory
+    default: '.'
+runs:
+  using: composite
+  steps:
+    - name: Set up Pandoc for vignette building
+      uses: r-lib/actions/setup-pandoc@v2
+
+    - name: Set up R ${{ inputs.r-version }}
+      uses: r-lib/actions/setup-r@v2
+      with:
+        r-version: ${{ inputs.r-version }}
+        use-public-rspm: true
+
+    - name: Set up R dependencies
+      uses: r-lib/actions/setup-r-dependencies@v2
+      with:
+        extra-packages: any::rcmdcheck
+        needs: check
+
+    - name: Run package checks
+      if: inputs.check == 'true'
+      uses: r-lib/actions/check-r-package@v2
```

Comment:
Right now, this is a totally self-contained action that also sets up R, dependencies, and optionally runs tests. Would any users want just the part that publishes to Package Manager to insert into their existing test workflow? 

I imagine most GHA users have something like [check-standard.yaml](https://github.com/r-lib/actions/blob/v2/examples/check-standard.yaml) that already sets up R, dependencies/pandoc, and runs package tests. These users might want to add a publish step at the end, for the `main` branch only, so they don't have duplicate test workflows for each R version/OS combination.

I think r-lib/actions has some actions like this, that assume you already have R and dependencies installed, and show the full setup through an example action. We could do the same, or provide both full and publish-only versions of the action. If it's possible in GHA, the full action could reuse the publish-only action for the publishing part.

Reply:
@glin I took a look at splitting these apart, and it makes things a lot messier and harder to follow.  I'm thinking maybe for now we leave it just as the build + publish part, and see if there's much demand for a publish-only step later, if that's ok?

Reply:
Sure, that sounds good.